### PR TITLE
Antismash42

### PR DESCRIPTION
**Pull request by zachcp**
* [ ] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

Version bump of antismash. 

Use new download location, remove unneeded patchfile and ditch perl as the code that required perl has been ported to python (although perl still seems to be on the dspendency chain....)

### CODE REVIEW

File: recipes/antismash/post-link.sh [hunk 1]
```diff
@@ -1,4 +0,0 @@
-#!/bin/bash
-
-# download the databases required by several options
-#$PREFIX/bin/download-antismash-databases
```
Comment (bgruening) [inline recipes/antismash/post-link.sh]: Are you sure it is not doing something?

**Comment by zachcp** [comment]
this downloads the databases. The initial recipe didnt't have this and left
it to the user. I can restore it once I get the builds running again.
Currently there is an import error for the primary antismash script.

**Comment by bgruening** [comment]
ping @kblin 
@zachcp are you going to try the rebuild of the missing packages?

**Comment by zachcp** [comment]
I updated fasttree and will take a stab at some of the other packages, time permitting.

**Comment by kblin** [comment]
Thanks for the quick fix on the hmmer3 version. :+1:

**Comment by zachcp** [comment]
@kblin - no prob - would love to see this built and get a conda build for your work on AS5.

@bgruening do you know if there  is a way to force conda to list the packages that need a rebuild?  The error messages are fairly opaque. The number of errors has been reducing over time as the package backlog moves through but it would be nice to focus on the particular offenders we care about. 

```
Build prefix failed with prefix length 255
WARNING:conda_build.environ:Build prefix failed with prefix length 255
Error was: 
WARNING:conda_build.environ:Error was: 
Placeholder of length '80' too short in package /Users/distiller/project/miniconda/conda-bld/antismash_1533574355556/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/bin/alphtype.
The package must be rebuilt with conda-build > 2.0.
WARNING:conda_build.environ:Placeholder of length '80' too short in package /Users/distiller/project/miniconda/conda-bld/antismash_1533574355556/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/bin/alphtype.
The package must be rebuilt with conda-build > 2.0.
One or more of your package dependencies needs to be rebuilt with a longer prefix length.
WARNING:conda_build.environ:One or more of your package dependencies needs to be rebuilt with a longer prefix length.
Falling back to legacy prefix length of 80 characters.
WARNING:conda_build.environ:Falling back to legacy prefix length of 80 characters.
Your package will not install into prefixes > 80 characters.
WARNING:conda_build.environ:Your package will not install into prefixes > 80 characters.
```

**Comment by bgruening** [comment]
@zachcp its most likely meme. If you can take this one over https://github.com/bioconda/bioconda-recipes/pull/10334 that would be great.

**Comment by zachcp** [comment]
@kblin can you take a quick look at this.  At the end of the day this was a small update that took a long time because of a major update in the build system.  Also, I can help field bioconda related issues for antismash builds

File: recipes/antismash/meta.yaml [hunk 2]
```diff
@@ -37,31 +33,31 @@ requirements:
...
     - helperlibs
-    - hmmer
+    - hmmer ==3.1b2
     - hmmer2
     - openjdk
     - libxml2 ==2.9.4
     - mafft
     - matplotlib
-    - meme ==4.11.2
+    - meme ==4.12.0
```
Comment (kblin) [inline recipes/antismash/meta.yaml]: I'd have to check `CASSIS` still works with that version of meme. I seem to recall that they changed the output file format from 4.11.3 to 4.11.4, and that broke some things. I don't remember if we got around to fixing it in a version-agnostic manner. I'll get back to you on that.
Is there a particular reason you're not leaving it pinned to 4.11.2, which definitely works?

File: recipes/antismash/meta.yaml [hunk 3]
```diff
@@ -37,31 +33,31 @@ requirements:
...
     - pandas
     - perl-bioperl
-    - perl-pathtools
     - prodigal
     - pyexcelerator
     - pyquery ==1.2.9
     - pysvg
     - scipy
-    - scikit-learn ==0.18.2
+    - scikit-learn
```
Comment (kblin) [inline recipes/antismash/meta.yaml]: RODEO comes with scikit-learn models trained and pickled with that version of scikit-learn, IIRC. Not sure how those models load back into the current 0.19 series.

**Comment by kblin** [comment]
I forgot to mention, thanks a lot for working on all of this, @zachcp !

**Comment by zachcp** [comment]
@kblin Thanks for the review. I'll retry with the pinned versions: the builds take a long time so I was going for the latest versions of packages that were likely to have been built most recently. I can back track on those two and see if it builds.  

update: rebuilding 4.11.2 in #11009

**Pull request merged**